src/inmanta/deploy/scheduler.py

```diff
@@ -736,10 +738,11 @@ async def _get_last_non_deploying_state_for_dependencies(
         :param resource: The id of the resource to find the dependencies for
         """
         requires_view: Mapping[ResourceIdStr, Set[ResourceIdStr]] = self._state.requires.requires_view()
-        dependencies: Set[ResourceIdStr] = requires_view.get(resource, set())
+        dependencies: Mapping[ResourceIdStr, ResourceState] = {
+            dep_id: self._state.resource_state[dep_id] for dep_id in requires_view.get(resource, set())
+        }
         dependencies_state = {}
-        for dep_id in dependencies:
-            resource_state_object: ResourceState = self._state.resource_state[dep_id]
+        for dep_id, resource_state_object in dependencies.items():
```

Comment:
Why this change? I think it has a slight performance impact, so I'd only do it with good reason. Do you find it more readable?